src/reflect/scala/reflect/internal/Kinds.scala

```diff
@@ -176,11 +172,11 @@ trait Kinds {
           // conceptually the same. Could also replace the types by
           // polytypes, but can't just strip the symbols, as ordering
           // is lost then.
-          val declaredBounds     = transform(hkparam.info.instantiateTypeParams(tparams, targs).bounds, paramowner)
-          val declaredBoundsInst = transform(bindHKParams(declaredBounds), owner)
-          val argumentBounds     = transform(hkarg.info.bounds, owner)
+          val declaredBounds     = hkparam.info.instantiateTypeParams(tparams, targs).bounds.asSeenFrom(pre, paramOwner)
+          val declaredBoundsInst = declaredBounds.substSym(underHKParams, withHKArgs).asSeenFrom(pre, owner)
+          val argumentBounds     = hkarg.info.bounds.asSeenFrom(argPre, argOwner).asSeenFrom(pre, owner)
```

Comment:
Outside of piping things, the only substantive change seems to be this new asSeenFrom invocation (right, @joroKr21?)

Reply:
Yes, that's right. Unfortunately both `asSeenFrom` seem necessary. They only work if you use the correct prefix and owner but here we don't know which one it is actually (could be either one).